batch/send-job.py

```diff
@@ -1,47 +1,146 @@
+    watershed_name: str
+    domain_name: str
+    duration: int
+    atlas_14_uri: str
+    job_def: str
+    job_queue: str
+    s3_bucket: str = "tempest"
+    por_start: datetime = datetime(1979, 2, 1, 0)
+    por_end: datetime = datetime(2022, 12, 31, 23)
+    scale: float = 12.0
+    watershed_uri: str = field(init=False)
+    watershed_s3_key: str = field(init=False)
+    domain_uri: str = field(init=False)
+    domain_s3_key: str = field(init=False)
+    output_prefix: str = field(init=False)
+
+    def __post_init__(self):
+        formatted_watershed_name = self.__format_name(self.watershed_name)
+        formatted_domain_name = self.__format_name(self.domain_name)
+
+        self.domain_s3_key = f"watersheds/{formatted_watershed_name}/{formatted_watershed_name}-transpo-area-{formatted_domain_name}.geojson"
+        self.domain_uri = f"s3://{self.s3_bucket}/{self.domain_s3_key}"
+
+        self.watershed_s3_key = f"watersheds/{formatted_watershed_name}/{formatted_watershed_name}.geojson"
+        self.watershed_uri = f"s3://{self.s3_bucket}/{self.watershed_s3_key}"
+        self.output_prefix = f"watersheds/{formatted_watershed_name}/{formatted_watershed_name}-transpo-area-{formatted_domain_name}/{self.duration}h"
+
+    @staticmethod
+    def __format_name(name: str) -> str:
+        cleaned = name.strip()
+        lower = cleaned.lower()
+        replaced = lower.replace(" ", "-")
+        return replaced
+
+    def get_job_name(self, dt: datetime) -> str:
+        job_name = f"{self.watershed_name.replace(' ','')}-{self.domain_name}-{self.duration}h-{dt.strftime('%Y%m%d')}"
+        return job_name
+
+
+def load_inputs(json_path: str) -> JobInput:
+    with open(json_path, "r") as f:
+        data = json.load(f)
+        if "por_start" in data.keys():
+            data["por_start"] = datetime.strptime(data["por_start"], "%Y-%m-%d %H:%M")
+        if "por_end" in data.keys():
+            data["por_end"] = datetime.strptime(data["por_end"], "%Y-%m-%d %H:%M")
+    return JobInput(**data)
+
+
+def check_exists(keys: list[str], bucket: str, client) -> None:
+    for key in keys:
+        print(f"Checking {key}")
+        logging.info(f"Checking {key}")
+        client.head_object(Bucket=bucket, Key=key)
+
+
+def construct_command(job_input: JobInput, current_dt: datetime) -> list[str]:
+    cmd_list = [
+        "python3",
+        "extract_storms_v2.py",
+        "-s",
+        current_dt.strftime("%Y-%m-%d"),
+        "-hr",
+        str(job_input.duration),
+        "-w",
+        job_input.watershed_name,
+        "-wu",
+        job_input.watershed_uri,
+        "-d",
+        job_input.domain_name,
+        "-du",
+        job_input.domain_uri,
+        "-b",
+        job_input.s3_bucket,
+        "-p",
+        job_input.output_prefix,
+        "-a",
+        job_input.atlas_14_uri,
+        "-sm",
+        str(job_input.scale),
+    ]
+    return cmd_list
+
+
+def send(job_input: JobInput) -> None:
+    # Create batch client
```

Comment:
These kinds of comments are not super helpful as they just explain what you are doing in the code, not why you are doing it.